src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorProjectChangePublisher.cs

```diff
@@ -209,9 +209,12 @@ protected virtual void DeleteFile(string publishFilePath)
 
         protected virtual void SerializeToFile(ProjectSnapshot projectSnapshot, string publishFilePath)
         {
+            var stringWriter = new StringWriter();
+            _serializer.Serialize(stringWriter, projectSnapshot);
```

Comment:
If things work without this string writer piece I'd rather not have it because we'll be allocating massive strings in VS